### PR TITLE
bottom-rs: init at unstable-2021-04-27

### PR DESCRIPTION
**Pull request by winterqt**
###### Motivation for this change

[`bottom-rs`](https://github.com/bottom-software-foundation/bottom-rs) is a comedic command line tool, similar to `cowsay` and the like. It doesn't really serve any purpose other than to 💖✨✨✨✨🥺,,,👉👈💖💖,👉👈✨✨✨,,👉👈💖💖,,👉👈💖💖✨🥺,,👉👈💖💖✨👉👈💖💖✨👉👈💖💖✨✨,👉👈 (`bottomify --regress ...`)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/bottom-rs/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, rustPlatform, fetchFromGitHub, }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "bottom-rs";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "bottom-software-foundation";
+    repo = pname;
+    rev = "3451cdadd7c4e64fe8e7f43e986a18628a741dec";
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/bottom-rs/default.nix:10]: See https://nixos.org/manual/nixpkgs/stable/#sec-package-naming:
`If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".`

File: pkgs/tools/misc/bottom-rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "0kr18q80021s1n9zzzff6w6yvbbjnk6zbbabi5b42b0rfv6fnfn2";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
+  cargoSha256 = "1cmz16882dfzsydxwr8glmd68vdqi3c8f11297dppdnw8lzxin95";
+
+  meta = with lib; {
+    description =
+      "Fantastic (maybe) CLI for translating between bottom and human-readable text";
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/bottom-rs/default.nix:19]: No need for this to be on multiple lines, IMO.

File: pkgs/tools/misc/bottom-rs/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ lib, rustPlatform, fetchFromGitHub, }:
```
Comment (IvarWithoutBones) [inline pkgs/tools/misc/bottom-rs/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub }:
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
A few nitpicks, but otherwise LGTM :+1:

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120937` at 3376b654 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/cdaf541ae122a04a43c557049df48e73">bottom-rs</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by IvarWithoutBones** [comment]
`bottom-rs-unstable-2021-04-27-vendor.tar.gz` seems to produce results with different checksums on the same exact commit, differing from run to run. Unsure why this would occur, but the package seems to be undeterministic.

**Comment by winterqt** [comment]
> but the package seems to be undeterministic.

As far as I can tell, the contents of the archives themselves are identical, so unsure why this would be happening...
```
> diff -r a/ b/
<no output>
```

**Comment by lukegb** [comment]
I think your last rebase went awry; mind removing the unrelated commits?

**Comment by winterqt** [comment]
@lukegb Should be good now.

**Comment by lukegb** [comment]
This doesn't actually seem to build - the FOD hashes are wrong?

**Comment by winterqt** [comment]
See https://github.com/NixOS/nixpkgs/pull/120937#issuecomment-827963063 and the comment directly below it. I still have no clue what is going on there, I'm guessing something upstream? Not exactly sure though...

File: pkgs/tools/misc/bottom-rs/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  cargoPatches = [ ./cargo-lock.patch ];
+  cargoSha256 = "1kgm8lc92lrg09fa2l2ajp92jkfn5n4glpgrvh6xyhhzd4pmgd9w";
+
+  meta = with lib; {
+    description = "Fantastic (maybe) CLI for translating between bottom and human-readable text";
+    homepage = "https://github.com/bottom-software-foundation/bottom-rs";
+    license = licenses.mit;
+    maintainers = with maintainers; [ winterqt ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bottom-rs/default.nix:22]:
```suggestion
```

File: pkgs/tools/misc/bottom-rs/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "bottom-software-foundation";
+    repo = pname;
+    rev = "3451cdadd7c4e64fe8e7f43e986a18628a741dec";
+    sha256 = "0kr18q80021s1n9zzzff6w6yvbbjnk6zbbabi5b42b0rfv6fnfn2";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
+  cargoSha256 = "1kgm8lc92lrg09fa2l2ajp92jkfn5n4glpgrvh6xyhhzd4pmgd9w";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bottom-rs/default.nix:15]:
```suggestion
  cargoSha256 = "0nyzg6pg69bf9vvc3r5lnhmkb9s1508c1gqcra3y43zffdlwml1y";
```

**Comment by IvarWithoutBones** [comment]
> > Do you have any clue why this would be happening?
> 
> Do you use the latest rust which is in master? It changed the hashes of a lot of vendor directories.
> 
> Work on ofborg and locally, so merging.

@SuperSandro2000 As discussed above, build failures occur due to hash mismatches even if nothing has changed. I can run `nixpkgs-review` on this PR twice, and it will generate a different hash every single time. As the author of this PR stated, this fails to compile on their system as well. I think we should not merge packages which undeterministically fail to compile on multiple systems to master.

Also, the commits you have added do not seem to be following nixpkgs guidelines?

**Pull request merged**